File: backend/src/main/java/br/ufpr/sept/so2/modules/publico/api/PublicoController.java
package br.ufpr.sept.so2.modules.publico.api;

import br.ufpr.sept.so2.shared.config.ContatoProperties;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/publico")
@Tag(name = "Público", description = "Endpoints sem autenticação (RF-F0-004)")
public class PublicoController {

    private final ContatoProperties contatoProperties;

    public PublicoController(ContatoProperties contatoProperties) {
        this.contatoProperties = contatoProperties;
    }

    @GetMapping("/contato")
    @Operation(summary = "Informações de contato da secretaria")
    public ContatoResponse contato() {
        return new ContatoResponse(
                contatoProperties.getNome(),
                contatoProperties.getEndereco(),
                contatoProperties.getTelefone(),
                contatoProperties.getEmail(),
                contatoProperties.getHorario()
        );
    }

    public record ContatoResponse(
            String nome,
            String endereco,
            String telefone,
            String email,
            String horario
    ) {
    }
}

File: backend/src/main/kotlin/br/ufpr/sept/so2/modules/publico/api/PublicoController.kt
package br.ufpr.sept.so2.modules.publico.api

import br.ufpr.sept.so2.shared.config.ContatoProperties
import io.swagger.v3.oas.annotations.Operation
import io.swagger.v3.oas.annotations.tags.Tag
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/publico")
@Tag(name = "Público", description = "Endpoints sem autenticação (RF-F0-004)")
class PublicoController(
    private val contatoProperties: ContatoProperties,
) {

    @GetMapping("/contato")
    @Operation(summary = "Informações de contato da secretaria")
    fun contato(): ContatoResponse =
        ContatoResponse(
            contatoProperties.nome,
            contatoProperties.endereco,
            contatoProperties.telefone,
            contatoProperties.email,
            contatoProperties.horario,
        )

    data class ContatoResponse(
        val nome: String?,
        val endereco: String?,
        val telefone: String?,
        val email: String?,
        val horario: String?,
    )
}
